src/core/src/op/variadic_split.cpp

```diff
@@ -2,33 +2,69 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
-#include "ngraph/op/variadic_split.hpp"
+#include "openvino/op/variadic_split.hpp"
 
 #include <numeric>
 
 #include "bound_evaluate.hpp"
 #include "compare.hpp"
 #include "itt.hpp"
-#include "ngraph/validation_util.hpp"
+#include "openvino/core/validation_util.hpp"
 #include "openvino/reference/slice.hpp"
 #include "variadic_split_shape_inference.hpp"
 
-using namespace std;
-using namespace ngraph;
+namespace ov {
+namespace op {
+namespace variadic_split {
+namespace {
 
-op::v1::VariadicSplit::VariadicSplit(const Output<Node>& data,
-                                     const Output<Node>& axis,
-                                     const Output<Node>& split_lengths)
-    : Op({data, axis, split_lengths}) {
-    constructor_validate_and_infer_types();
+bool has_axis_and_splits_bound_set(const Node* const node) {
+    return have_node_inputs_bounds_set(node, 1, 2);
 }
 
-bool ngraph::op::v1::VariadicSplit::visit_attributes(AttributeVisitor& visitor) {
-    OV_OP_SCOPE(v1_VariadicSplit_visit_attributes);
+bool evaluate(TensorVector& outputs, const TensorVector& inputs) {
+    const auto& data_tensor = inputs[0];
+    const auto& axis_tensor = inputs[1];
+    const auto axis =
+        ov::util::normalize(get_tensor_data_as<int64_t>(axis_tensor).front(), data_tensor.get_shape().size());
+
+    ov::Coordinate upper_bounds(data_tensor.get_shape());
+    ov::Coordinate lower_bounds(upper_bounds.size());
+    upper_bounds[axis] = 0;
+
+    const Strides default_strides(upper_bounds.size(), 1);
+    constexpr auto is_zero_dim = ov::cmp::Equal<size_t>(0);
+
+    for (auto& output : outputs) {
+        const auto& out_shape = output.get_shape();
+        upper_bounds[axis] += out_shape[axis];
+
+        if (std::none_of(out_shape.cbegin(), out_shape.cend(), is_zero_dim)) {
+            reference::slice(static_cast<const char*>(data_tensor.data()),
+                             static_cast<char*>(output.data()),
+                             data_tensor.get_shape(),
+                             lower_bounds,
+                             upper_bounds,
+                             default_strides,
+                             out_shape,
+                             data_tensor.get_element_type().size());
+        }
+
+        lower_bounds[axis] = upper_bounds[axis];
```

Comment:
Is this assignment needed in every iteration?

Reply:
Yes is like  `new start` for next iteration.